Source/Particles/Collision/BinaryCollision/BinaryCollision.H

```diff
@@ -199,6 +217,10 @@ public:
                     amrex::HostDevice::Atomic::Add( &(*cost)[mfi.index()], wt);
                 }
             }
+
+            // Call redistribute to remove particles with negative ids
+            species1.Redistribute(lev, lev, 0, true, true);
+            species2.Redistribute(lev, lev, 0, true, true);
```

Comment:
Address https://github.com/ECP-WarpX/WarpX/pull/4577#discussion_r1531167214
```suggestion
            if (m_have_product_species) {
                // Call redistribute to remove particles with negative ids
                species1.Redistribute(lev, lev, 0, true, true);
                species2.Redistribute(lev, lev, 0, true, true);
            }
```